text/0000-grid-primitives.md

```diff
@@ -0,0 +1,760 @@
+boolean type assists in capturing intent and restricting the value.
+
+### Strings
+
+A string data type contains a standard UTF-8 encoded string value.
+
+### Numbers
+
+Numbers are represented as an integer with a given precision.  This can be
+thought of as akin to scientific notation. An instance of a number with this
+property definition is represented as a value (the significand) with the
+exponent (the order of magnitude) defined in the schema itself. So for example:
+
+```
+(value: 24, exponent: 3)  -> 24 * 10^3  -> 24000
+(value: 24, exponent: -3) -> 24 * 10^-3 -> 0.024
+(value: 24, exponent: 0)  -> 24 * 10^0  -> 24
+```
+
+Importantly, this exponent will be set on a Property's schema, not when the
+value is actually input. It will affect the semantic meaning of integers stored
+under a Property, not any of the actual operations done with them. Properties
+with an exponent of 3 or -3 will always be expressed as a whole integer of
+thousands or thousandths. For this reason, the exponent should be thought of
+more as a unit of measure than as true scientific notation. 
+
+Standard integers are represented with the exponent set to zero. 
+
+### Enums
+
+An enum data type restricts values to a limited set of possible values. The
+definition for this data type includes a list of strings names describing a
+possible state of the enum. 
+
+### Structs
+
+A struct is a recursively defined collection of other named properties that
+represents two or more intrinsically linked values, like X/Y coordinates or RGB
+colors. These values can be of any Grid primitive data type, including STRUCT,
+allowing nesting to an arbitrary depth. Although versatile and powerful, structs
+are heavyweight and should be used conservatively; restrict struct use to
+linking values that must always be updated together. The smart contract will
+enforce this usage, rejecting any transactions that do not have a value for
+every property in a struct.
+
+Note that although structs are built using a list of PropertyDefinitions, any
+nested use of the required property is meaningless and will be rejected by the
+smart contract. As Properties are set in their entirety, either all of the
+struct is required or none of it is. In other words, partial structs are not
+allowed.
+
+### Latitude/Longitude
+
+Latitude/Longitude (Lat/Long) values are represented as a pre-defined struct
+made up of a latitude, longitude pair.  Both latitude and longitude are
+represented as signed integers indicating millionths of degrees.
+
+### Datetime
+
+Datetime values are represented as an
+[ISO-8601](https://en.wikipedia.org/wiki/ISO_8601) combined date-time formats.
+For example, "2007-04-05T14:30Z" or "2007-04-05T12:30-02:00".
+
+## Schemas
+
+Property definitions are collected into a Schema data type, which defines all
+the possible properties for an item that belongs to a given schema.  These
+schemas include the following:
+
+- a name
```

Comment:
- **name** - the name and identifier for the Schema
- **description** - description of the Schema
- **owner** - the Pike compatible org identifier for the owning org of the Schema
- **properties** - a list of `PropertyDefinitions` that define the attributes of the entity being defined by the Schema